mobile/src/main/java/org/openhab/habdroid/core/OnUpdateBroadcastReceiver.java

```diff
@@ -0,0 +1,40 @@
+package org.openhab.habdroid.core;
+
+import android.content.BroadcastReceiver;
+import android.content.Context;
+import android.content.Intent;
+import android.content.SharedPreferences;
+import android.preference.PreferenceManager;
+import android.util.Log;
+
+import org.openhab.habdroid.util.Constants;
+
+import static org.openhab.habdroid.util.Constants.PREFERENCE_COMPAREABLEVERSION;
+
+public class OnUpdateBroadcastReceiver extends BroadcastReceiver {
+    private static final String TAG = OnUpdateBroadcastReceiver.class.getSimpleName();
+
+    private static final int UPDATE_LOCAL_CREDENTIALS = 25;
+
+    @Override
+    public void onReceive(Context context, Intent intent) {
+        if (!Intent.ACTION_MY_PACKAGE_REPLACED.equalsIgnoreCase(intent.getAction())) {
```

Comment:
equals() should be fine here, intent actions are case sensitive.

Reply:
+1